extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/security/RolesAllowedResource.java

```diff
@@ -24,4 +26,12 @@ public String admin() {
         return "admin";
     }
 
+    @NonBlocking
```

Comment:
@michalvavrik Can you clarify please, how does adding `@NonBlocking` is related to the fix, sorry of it is obvious, the affected code is using the proactive authentication flag to decide if the identity should be set only.

Reply:
@sberyozkin when the endpoint is annotated with `@NonBlocking` then it's running on IO Thread and that leads to the `BlockingOperationNotAllowedException` when you are blocking. Without `@NonBlocking` there should be no problems...

Reply:
so the fix should be necessary

Reply:
I don't think there is an issue in setting identity for blocking cases too as there should be no difference, but I can work around this if necessary.

Reply:
So can you please add a test method checking what happens without `@NonBlocking` ?

Reply:
I just found there are already tests that checks other scenarios (except for the `@NonBlocking`), should be sufficiently covered now.